### ISSUE TITLE
Is this a real-time method of action recognition?

### ISSUE THREAD

**Issue by JoshaPan** [kenziyuliu/MS-G3D#10]

**Comment by kenziyuliu** [comment]
Hi there,

The algorithm itself is designed to take the entire skeleton sequence at once (tensors in shape (channels, frames, nodes)) and make a prediction of the action class. You can certainly extend this repo to build a real-time recognition system e.g. by keeping a running buffer of skeletons and predict action based on the buffer at each time step.

**Comment by ywang0218** [comment]
> Hi there,
> 
> The algorithm itself is designed to take the entire skeleton sequence at once (tensors in shape (channels, frames, nodes)) and make a prediction of the action class. You can certainly extend this repo to build a real-time recognition system e.g. by keeping a running buffer of skeletons and predict action based on the buffer at each time step.

Can I use this model to predict action given rgb picture/video or skeleton data from openpose? Do you have a detection code/command for that?

**Comment by kenziyuliu** [comment]
Hi @ywang0218,

Thanks for your interest! This repo only provides the training/testing code for the public benchmark datasets (NTU RGB+D 60, NTU RGB+D 120, Kinetics Skeleton 400). 

For prediction on RGB videos/skeleton data from OpenPose **using pre-trained model**, I would suggest the following steps:
1.  Inspect the [original Kinetics dataset](https://github.com/yysijie/st-gcn/blob/master/OLD_README.md#kinetics-skeleton); these are the skeletons already extracted with OpenPose from Kinetics 400
2. Add some code to convert your own RGB videos into skeletons using OpenPose; make sure the format matches the Kinetics Skeleton dataset
3. [Normalize them and convert them to tensor format](https://github.com/kenziyuliu/MS-G3D/blob/master/data_gen/kinetics_gendata.py), and feed them to the pretrained model
4. For real-time video feed, you can use the same steps above by keeping a running buffer of the skeleton frames

If you want to train a new model from scratch for your own data, you can just follow the training workflow for Kinetics Skeleton.

Hope this helps :)
